meshtastic/tests/test_node.py

```
@@ -61,6 +61,24 @@ def __exit__(self, exc_type: Any, exc: Any, tb: Any) -> Literal[False]:
+class _TrackingLock:
+    """Lock stub that records how many times it was acquired."""
+
+    def __init__(self, on_exit: Callable[[], None] | None = None) -> None:
+        self.enter_count = 0
+        self._on_exit = on_exit
+
+    def __enter__(self) -> "_TrackingLock":
+        self.enter_count += 1
+        return self
+
+    def __exit__(self, exc_type: Any, exc: Any, tb: Any) -> Literal[False]:
```

Ruff (0.15.5)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `exc_type`

(ANN401)